project/src/main/java/com/google/sps/utilities/SpannerTasks.java

```diff
@@ -128,7 +128,8 @@ public static void insertorUpdateEvent(Event event) {
                 Statement.of(
                     String.format(
                         "SELECT EventID, Name, Description, Labels, Location, Date, Time,"
-                            + " Host, Opportunities, Attendees, Image FROM %s WHERE EventID in (%s)",
+                            + " Host, Opportunities, Attendees FROM %s WHERE EventID in (%s)"
+                            + " AND DATE_DIFF(Date, CURRENT_DATE(), DAY) > 0",
```

Comment:
I can think of one improvement, you don't have to make it in this PR.

You can create a AutoValue, which includes all the possible conditions. And use the AutoValue to control what condition to add in the SQL, so you don't have to rewrite the condition every time.

For example

EventTableConditionParams {
  Optional<List> eventIds;
  Optional<boolean> isFuture;
}

function generateCondition(EventTableConditionParams params) {
  condition = ""
  params.eventIds.ifPresent(condition + EventID in (%s));
  params.isFuture.ifPresent(condtion + DATE_DIFF(Date, CURRENT_DATE(), DAY) > 0)
}